proxy/src/control/destination/background.rs

```diff
@@ -84,26 +112,34 @@ pub(super) fn task(
 {
     // Build up the Controller Client Stack
     let mut client = host_and_port.map(|host_and_port| {
-        let scheme = http::uri::Scheme::from_shared(Bytes::from_static(b"http")).unwrap();
-        let authority = http::uri::Authority::from(&host_and_port);
-        let connect = Timeout::new(
-            LookupAddressAndConnect::new(host_and_port.clone(), dns_resolver.clone(),
-                                         controller_tls),
-            Duration::from_secs(3),
-        );
-
-        let log = ::logging::admin().client("control", host_and_port.clone());
-        let h2_client = tower_h2::client::Connect::new(
-            connect,
-            h2::client::Builder::default(),
-            log.executor()
+        // Destructure `controller_tls` to separate the `ClientConfigWatch`
+        // from our `tls::Identity`.
+        let (identity, watch) = match controller_tls {
+            Conditional::Some(tls::ConnectionConfig {
+                identity,
+                config: watch
+            }) => {
+                (Conditional::Some(identity), watch)
+            },
+            Conditional::None(reason) => {
+                // If there's no connection config, then construct a new
+                // `Watch` that never updates to construct the `WatchService`.
+                // In this case, it seems unnecessary to wrap the stack in a
```

Comment:
I'd drop everything after the first sentence. It makes the block _less_ clear imo